ydb/docs/en/core/integrations/orm/jooq.md

```diff
@@ -0,0 +1,187 @@
+# JOOQ Extension for use with {{ ydb-short-name }}
+
+This guide is intended for use with [JOOQ](https://www.jooq.org/) and {{ ydb-short-name }}.
+
+JOOQ is a Java library that allows you to create type-safe SQL queries by generating Java classes from a database schema and using convenient query builders.
+
+## Generating Java Classes {#generated-java-classes}
+
+You can generate Java classes using any tools provided on the [official JOOQ website](https://www.jooq.org/doc/latest/manual/code-generation/codegen-configuration/). Two dependencies are required: {{ ydb-short-name }} JDBC driver and JOOQ extension for {{ ydb-short-name }}, as well as two parameters:
+
+- `database.name`: `tech.ydb.jooq.codegen.YdbDatabase` (mandatory setting)
+- `strategy.name`: `tech.ydb.jooq.codegen.YdbGeneratorStrategy` (recommended setting)
+
+Let's look at an example of the `maven` plugin:
+
+```xml
+<plugin>
+    <groupId>org.jooq</groupId>
+    <artifactId>jooq-codegen-maven</artifactId>
+    <version>3.19.11</version>
+    <executions>
+        <execution>
+            <goals>
+                <goal>generate</goal>
+            </goals>
+        </execution>
+    </executions>
+    <dependencies>
+        <dependency>
+            <groupId>tech.ydb.jdbc</groupId>
+            <artifactId>ydb-jdbc-driver</artifactId>
+            <version>${ydb.jdbc.version}</version>
+        </dependency>
+        <dependency>
+            <groupId>tech.ydb.dialects</groupId>
+            <artifactId>jooq-ydb-dialect</artifactId>
+            <version>${jooq.ydb.version}</version>
+        </dependency>
+    </dependencies>
+    <configuration>
+        <jdbc>
+            <driver>tech.ydb.jdbc.YdbDriver</driver>
+            <url>jdbc:ydb:grpc://localhost:2136/local</url>
+        </jdbc>
+        <generator>
+            <strategy>
+                <name>tech.ydb.jooq.codegen.YdbGeneratorStrategy</name>
+            </strategy>
+            <database>
+                <name>tech.ydb.jooq.codegen.YdbDatabase</name>
+                <!-- исключение системных таблицы -->
+                <excludes>.sys.*</excludes>
+            </database>
+            <target>
+                <packageName>ydb</packageName>
+                <directory>./src/main/java</directory>
+            </target>
+        </generator>
+    </configuration>
+</plugin>
+```
+
+Example of generated classes from [YQL tutorial](../../dev/yql-tutorial/create_demo_tables.md):
```

Comment:
It'd be helpful to post the full classes somewhere on GitHub and give a link to them here.